src/mpid/ch4/shm/posix/eager/fbox/fbox_init.h

```diff
@@ -41,7 +59,17 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_eager_init(int rank, int size)
     MPIR_CHKPMEM_DECL(5);
 
     MPIDI_POSIX_eager_fbox_control_global.num_seg = 1;
-    MPIDI_POSIX_eager_fbox_control_global.next_poll_local_rank = 0;
+    MPIDI_POSIX_eager_fbox_control_global.first_poll_local_ranks =
+        MPL_malloc(sizeof(*MPIDI_POSIX_eager_fbox_control_global.first_poll_local_ranks) *
+                   MPIR_CVAR_CH4_POSIX_EAGER_FBOX_POLL_CACHE_SIZE + 1, MPL_MEM_SHM);
+    if (NULL == MPIDI_POSIX_eager_fbox_control_global.first_poll_local_ranks)
+        MPIR_ERR_SET(mpi_errno, MPI_ERR_OTHER, "**nomem");
+
+    /* -1 means we aren't looking for anything in particular. */
+    for (i = 0; i < MPIR_CVAR_CH4_POSIX_EAGER_FBOX_POLL_CACHE_SIZE; i++) {
+        MPIDI_POSIX_eager_fbox_control_global.first_poll_local_ranks[i] = -1;
+    }
+    MPIDI_POSIX_eager_fbox_control_global.first_poll_local_ranks[i] = 0;
 
```

Comment:
Why do you set `MPIDI_POSIX_eager_fbox_control_global.first_poll_local_ranks[i] = 0;` after initialization ? I see the comment at `MPIDI_POSIX_eager_fbox_control_t `. It might be better if a short comment can be also added here.

Reply:
Good point. 👍 I'll mention that this box is special.